http/typescript/fetch/src/httpCore.ts

```diff
@@ -3,98 +3,118 @@ import { Headers as FetchHeadersCtor } from 'cross-fetch';
 import { ReadableStream } from 'web-streams-polyfill';
 import { URLSearchParams } from 'url';
 import { HttpClient } from './httpClient';
+import { Middleware } from "./middleware";
+import { FetchOptions } from './fetchOptions';
+
+export interface RequestBuilderConfig {
+    parseNodeFactory: ParseNodeFactory,
+    serializationWriterFactory: SerializationWriterFactory
+}
+
+export interface CoreConfig {
+    authenticationProvider: AuthenticationProvider,
+    httpClient?: HttpClient,
+    middleware?: Middleware | Middleware[],
+    fetchOptions?: FetchOptions
+}
+
+interface CoreOptions {
+    authenticationProvider: AuthenticationProvider,
+    httpClient: HttpClient,
+    fetchOptions?: FetchOptions
+}
+
 export class HttpCore implements IHttpCore {
     public getSerializationWriterFactory(): SerializationWriterFactory {
-        return this.serializationWriterFactory;
+        return this.requestBuilderConfig.serializationWriterFactory;
     }
     /**
+     * @private
+     * @constructor
      * Instantiates a new http core service
-     * @param authenticationProvider the authentication provider to use.
-     * @param parseNodeFactory the parse node factory to deserialize responses.
-     * @param serializationWriterFactory the serialization writer factory to use to serialize request bodies.
-     * @param httpClient the http client to use to execute requests.
      */
-    public constructor(public readonly authenticationProvider: AuthenticationProvider, private parseNodeFactory: ParseNodeFactory = ParseNodeFactoryRegistry.defaultInstance, private serializationWriterFactory: SerializationWriterFactory = SerializationWriterFactoryRegistry.defaultInstance, private readonly httpClient: HttpClient = new HttpClient()) {
-        if(!authenticationProvider) {
+    private constructor(private requestBuilderConfig: RequestBuilderConfig = { parseNodeFactory: ParseNodeFactoryRegistry.defaultInstance, serializationWriterFactory: SerializationWriterFactoryRegistry.defaultInstance }, private coreOptions: CoreOptions) {
+        if (!coreOptions.authenticationProvider) {
```

Comment:
the configuration objects should also be tested before those properties are tested for

Reply:
The methods and the logic of this class introduced in this PR changes are temporary. The idea of this PR is to introduce the config object pattern and the factory methods and to get the FetchOptions merged.

More changes in this class will be present in the upcoming PRs.